src/Parallel/Algorithm.hpp

```diff
@@ -201,6 +212,10 @@ class AlgorithmImpl<ParallelComponent, tmpl::list<PhaseDepActionListsPack...>>
     p | inboxes_;
     p | array_index_;
     p | global_cache_;
+    // note that `invoke_component_pup_with_current_box` passes the `box_` by
+    // const reference. If mutable access is required to the box, this function
+    // call needs to be carefully considered with respect to the `p | box_` call
+    // in both packing and unpacking scenarios.
     invoke_component_pup_with_current_box(p, box_);
```

Comment:
can we rename this function to something like `perform_registration_or_deregistration` to reflect what it is actually doing (currently)?